### ISSUE TITLE
Enchance the mod by exposing/making some events we can listen to

### ISSUE THREAD

**Issue by MxOrbulent** [Additional-Mods/simplerauth#9]
**Is your feature request related to a problem? Please describe.**
No not related to a problem.

**Describe the solution you'd like**
Add 3 events that this mod will fire that other mod authors can listen in on.
OnLoginEvent, OnRegisterEvent and OnLoginFailEvent.

The two first elements simply has the player, OnLoginFailEvent also contains the number of fails the player is at when the event last fired.

For example, while some might want to listen to OnLoginFailEvent to kick if someone fails to login 3 times, someone else might want to do that AND add a custom book to their inventory that tells the actual account owner that someone else has been trying to login but failed. 
OnRegisterEvent could be something like if someone wants to have a special room where unregistered people are sent to (by making the server default spawn point to that room) and when they register, the mod author can then decide where to send these people.

OnLoginEvent could be used for example to make additional changes a mod author would want after login.
For example, if someone spots someone about to login, until they have actually logged in, they are effectively helpless due to being frozen. 
A mod author could then add " if air below logging in player is more than 1 block deep and or is sorrounded by obsidian then teleport to random spot nearby that has free access to the sky and a non-lava/magma/tnt block below " which would make it harder to spawncamp."

Basically by exposing these 3 events, your mod can stay Simple for those who want it, while it unlocks a whole world of shenanigans for mod authors who want to significantly extend what's possible without you yourself having to write it, allowing you to focus on the mod itself.

There is some additional enchantments I was thinking off that are loose related to OnRegisterEvent. See that feature request here: https://github.com/Dqu1J/simplerauth/issues/10

**Comment by Dqu1J** [comment]
Was implemented in #36
